src/main.rs

```diff
@@ -290,6 +297,74 @@ fn execute_cli_mode(mut cli: Cli) -> Result<(), Diagnostics> {
         println!("{stats}");
       }
     }
+
+    Mode::RunC(RunArgs2 { pretty, run_opts, comp_opts, warn_opts, path, arguments }) => {
```

Comment:
Instead of duplicating this, you could match the three cases here and then inside match again to get the specific function you want to run.